mlos_bench/mlos_bench/optimizers/base_optimizer.py

```diff
@@ -317,27 +311,33 @@ def _get_score(self, status: Status, score: Optional[Union[float, Dict[str, floa
         ----------
         status : Status
             Final status of the experiment (e.g., SUCCEEDED or FAILED).
-        score : Union[float, Dict[str, float]]
-            A scalar or a dict with the final benchmark results.
+        scores : Optional[Dict[str, TunableValue]]
+            A dict with the final benchmark results.
             None if the experiment was not successful.
 
         Returns
         -------
-        score : float
-            A scalar benchmark score to be used as a primary target for MINIMIZATION.
+        score : Optional[Dict[str, float]]
+            An optional dict of benchmark scores to be used as targets for MINIMIZATION.
         """
         if not status.is_completed():
             return None
-        if status.is_succeeded():
-            assert score is not None
-            if isinstance(score, dict):
-                if self._opt_target not in score:
-                    raise ValueError(f"Missing expected optimization target metric '{self._opt_target}' "
-                                     + f"in results for iteration {self._iter}: {score}")
-                score = score[self._opt_target]
-            return float(score) * self._opt_sign
-        assert score is None
-        return float("inf")
+
+        if not status.is_succeeded():
+            assert scores is None
+            return {
+                opt_target: float("inf")
+                for opt_target in self._opt_targets.keys()
+            }
+
+        assert scores is not None
+        target_metrics: Dict[str, float] = {}
+        for (opt_target, opt_dir) in self._opt_targets.items():
+            val = scores[opt_target]
+            assert val is not None
```

Comment:
```suggestion
```

Maybe unnecessary based on latest type changes?

Reply:
Nah, the linter complains - `val` is of type `TunableValue` that theoretically can be `None`